parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -1148,12 +1148,17 @@ private void readAllPartsVectoredOrNormal(List<ConsecutivePartList> allParts, Ch
       throws IOException {
 
     if (shouldUseVectoredIO(allParts)) {
-      readVectored(allParts, builder);
-    } else {
-      for (ConsecutivePartList consecutiveChunks : allParts) {
-        consecutiveChunks.readAll(f, builder);
+      try {
+        readVectored(allParts, builder);
+        return;
+      } catch (IOException | IllegalArgumentException | UnsupportedOperationException e) {
+        // possible failure modes.
+        LOG.warn("readVectored() failed; falling back to normal IO", e);
```

Comment:
Do we need to support fallback? What if some parts were read successfully and failed in later stages?

Reply:
That is actually a valid point, but it gets complicated fast. Why don't you submit a follow-up PR? You will need to split out the successful from the unsuccessful reads and only execute the new ones.

Reply:
Yes, it will get complicated and that is why I was thinking is that an overkill or should we just fail in case of failure? 
In the meantime, I will try to see how complicated it gets code wise.